README.md

```diff
@@ -199,6 +200,74 @@ state, and shell history.
 successful package installation and working build. This is run in our development
 continuous delivery pipeline to ensure the package works as expected.
 
+### Debugging
+The`setup` subcommand has a `--debug` option which can be used to get detailed 
+output from all the commands run during the preconfigure process. This can be
+added to the Azure ARM template or AWS CloudFormation template as needed.
+
+#### Azure
+Logs from the ARM template script can be found here:
+`/var/log/azure/Microsoft.OSTCExtensions.CustomScriptForLinux/1.5.2.2/extension.log`
+
+The script itself can be found here:
+`/var/lib/waagent/Microsoft.OSTCExtensions.CustomScriptForLinux-1.5.2.2/download/0/`
+
+##### Adding debug option
+To add the `--debug` flag, you will need to modify the command executed here:
+https://github.com/chef-partners/omnibus-marketplace/blob/master/arm-templates/automate/automate_setup.rb#L45
+
+#### AWS
+We currently have to offerings in AWS which operate slightly differenly in the
+manner in which they are provisioned. The BYOL version uses CloudFormation,
+while the Metered offering includes a pre-baked init script. With the
+CloudFormation template, you can make the change and test with no other
+modifications or actions required.
+
+However, since the metered version has the scriped baked into the AMI,
+both a new version of the chef-marketplace and a new AMI build will be required 
+to be able to use the change.
+
+##### Adding the debug option
+###### BYOL
+To add the `--debug` flag, you will need to modify the CloudFormation template similar
+to the example below:
+
+```
+--- a/cloudformation/marketplace_byol.yml
++++ b/cloudformation/marketplace_byol.yml
+@@ -257,7 +257,7 @@ Resources:
+             - !If
+               - HasLicenseUrl
+               - !Sub >-
+-                chef-marketplace-ctl setup --preconfigure --license-url ${LicenseUrl}
++                chef-marketplace-ctl setup --preconfigure --license-url ${LicenseUrl} --debug 2>&1 >/var/log/cfn-userdata.log
```

Comment:
Not related to this change at all, but perhaps we should always enable debug to the file in the published template?

Reply:
@ryancragun There is some security issues with that since it will log private keys and other sensitive information to that file. I would continue to leave it out by default.